### PR TITLE
docs: fixup builds YAML config

### PR DESCRIPTION
**Pull request by gctucker**
Add and update the page about the YAML builds configuration.

### CODE REVIEW

**Comment by gctucker** [comment]
Tested OK on staging: https://static.staging.kernelci.org/docs/core/config/

File: doc/config.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+ - *trees* are git repositories
+ - *config fragments* are modular lists of kernel config values
+ - *build environments* define the available compilers
+ - *architectures* are to keep all the properties related to each CPU
+   architecture build variants are to define a set of builds done within
+   a build configuration
+ - *build configurations* are combinations of all of the above
+
+The build configurations are the main entries as they define what is
+actually going to be be built. They must have one tree and one branch
```
Comment (alexandrasp) [inline doc/config.md:26]: typo `to be be`

File: doc/config.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+defined, and optionally more details such as which compilers to use,
+which architectures and defconfigs to build etc... There should not be
```
Comment (alexandrasp) [inline doc/config.md:28]: maybe use just one dot after `etc`

File: doc/config.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+more than one build config with a given tree / branch combination to
+avoid monitoring the same branch more than once. Each build entry can
+have several build variants as sub-sections to define multiple ways of
+building a same branch.
+
+### How to add trees and branches
+
+Each tree is defined in the `trees` section. So to add a new tree, that is
```
Comment (alexandrasp) [inline doc/config.md:36]: section `trees` or `tree`?

File: doc/config.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  renesas:
+    tree: renesas
+    branch: 'devel'
+
+  renesas_next:
+    tree: renesas
+    branch: 'next'
+```
+
+The build configuration names typically follow the tree name and the
```
Comment (alexandrasp) [inline doc/config.md:59]: typo `names`

**Review by alexandrasp — CHANGES_REQUESTED** [review]

File: doc/config.md (hunk 3 above)
Comment (gctucker) [reply doc/config.md:36]: The YAML entry is `trees`.

File: doc/config.md (hunk 2 above)
Comment (gctucker) [reply doc/config.md:28]: Ah yes, that's an old habit.

File: doc/config.md (hunk 4 above)
Comment (gctucker) [reply doc/config.md]: I meant the name of each tree and branch.  The wording was a bit confusing anyway so I've rewritten it now.

**Review by alexandrasp — APPROVED** [review]
LGTM

**Pull request merged**